service/storage_service.cc

```diff
@@ -296,6 +296,21 @@ future<> storage_service::wait_for_ring_to_settle() {
     slogger.info("Checking bootstrapping/leaving nodes: ok");
 }
 
+static locator::node::state to_topology_node_state(node_state ns) {
+    switch (ns) {
+        case node_state::bootstrapping: return locator::node::state::bootstrapping;
+        case node_state::decommissioning: return locator::node::state::being_decommissioned;
+        case node_state::removing: return locator::node::state::being_removed;
+        case node_state::normal: return locator::node::state::normal;
+        case node_state::left_token_ring: return locator::node::state::left;
+        case node_state::left: return locator::node::state::left;
+        case node_state::replacing: return locator::node::state::replacing;
+        case node_state::rebuilding: return locator::node::state::normal;
```

Comment:
this caught my attention. shouldn't rebuilding be treated as joining state instead?

Reply:
Topology state load treats rebuilding nodes as normal:
```
            case node_state::rebuilding:
                // Rebuilding node is normal
                co_await add_normal_node(id, rs);
```

They are bootstrapped earlier, right?